terraform/node_provider.go

```diff
@@ -48,6 +48,13 @@ func (n *NodeApplyableProvider) ValidateProvider(ctx EvalContext, provider provi
 
 	configBody := buildProviderConfig(ctx, n.Addr, n.ProviderConfig())
 
+	// if provider config is empty, return early
```

Comment:
More explanation here in the comment would be nice -- why are we returning early? (explain this)

Suggestion, mashed up from the suggestions that led to this PR:

```
// if a provider config is empty (only an alias), return early and don't continue
// validation. validate doesn't need to fully configure the provider itself, so 
// skipping a provider with an implied configuration won't prevent other validation from completing.
```